### PR TITLE
Rework openCV test_packages to remove controversial Lena test image [1/2]

### PR DESCRIPTION
**Pull request by ericriff**
See https://github.com/conan-io/conan-center-index/issues/4609

This effort has been splitted in 2 PRs to workaround the CCI limitation of only modifying one recipe per PR. Another PR will fix 3.x.

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericriff** [comment]
Why does 2 version of the same library count as 2 separate libraries? :(

**Comment by prince-chrismc** [comment]
it's two different recipes

**Comment by Croydon** [comment]
Best case scenario would be a test that does not need a binary file in git

**Comment by ericLemanissier** [comment]
Conan-center-index should allow this situation, because it is really no different than modifying a single conanfile which builds several version. In both case there can be no requirement relation between the packages to build, so the order of build does not matter.

File: recipes/opencv/2.x/test_package/conanfile.py [hunk 1]
```diff
@@ -14,6 +14,6 @@ def build(self):
     def test(self):
         if not tools.cross_building(self.settings):
             bin_path = os.path.join("bin", "test_package")
-            lena = os.path.join(self.source_folder, "lena.jpg")
+            peppers = os.path.join(self.source_folder, "peppers.jpg")
             res_path = self.deps_cpp_info["opencv"].res_paths[0]
-            self.run("%s %s %s" % (bin_path, res_path,lena), run_environment=True)
+            self.run("%s %s %s" % (bin_path, res_path,peppers), run_environment=True)
```
Comment (mathbunnyru) [inline recipes/opencv/2.x/test_package/conanfile.py:19]:
```suggestion
            self.run("{} {} {}".format(bin_path, res_path, peppers), run_environment=True)
```

File: recipes/opencv/3.x/test_package/conanfile.py [hunk 2]
```diff
@@ -14,6 +14,6 @@ def build(self):
     def test(self):
         if not tools.cross_building(self.settings):
             bin_path = os.path.join("bin", "test_package")
-            lena = os.path.join(self.source_folder, "lena.jpg")
+            peppers = os.path.join(self.source_folder, "peppers.jpg")
             res_path = self.deps_cpp_info["opencv"].res_paths[0]
-            self.run("%s %s %s" % (bin_path, res_path,lena), run_environment=True)
+            self.run("%s %s %s" % (bin_path, res_path,peppers), run_environment=True)
```
Comment (mathbunnyru) [inline recipes/opencv/3.x/test_package/conanfile.py:19]:
```suggestion
            self.run("{} {} {}".format(bin_path, res_path, peppers), run_environment=True)
```

**Review by mathbunnyru — DISMISSED** [review]
Let's update the code a little bit, if you're changing the lines.

**Comment by ericriff** [comment]
So whatever approach we take, I have to split this into 2 PRs then?

**Comment by mathbunnyru** [comment]
Yep, I think the fastest way to get this merged is to create 2 different PRs.

**Review by mathbunnyru — APPROVED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
Please, remove png files and keep test more simple.

**Comment by ericriff** [comment]
I reverted the changes on 3.x (will open another PR later), got rid of the peppers image on 2.x and replaced the test_package with one similar to the one on 4.x, which doesn't use an image.

File: recipes/opencv/2.x/test_package/test_package.cpp [hunk 3]
```diff
@@ -1,67 +1,164 @@
...
-    face_cascade.detectMultiScale(frame_gray, faces, 1.1, 2, 0 | CASCADE_SCALE_IMAGE, Size(30, 30));
-
-    for (size_t i = 0; i < faces.size(); i++) {
-        Point center( faces[i].x + faces[i].width*0.5, faces[i].y + faces[i].height*0.5 );
-        ellipse( frame, center, Size( faces[i].width*0.5, faces[i].height*0.5), 0, 0, 360, Scalar( 255, 0, 255 ), 4, 8, 0 );
+#include <opencv2/core/core.hpp>
 
-        Mat faceROI = frame_gray( faces[i] );
-        std::vector<Rect> eyes;
+#define w 400
```
Comment (mathbunnyru) [inline recipes/opencv/2.x/test_package/test_package.cpp:10]: meh

**Review by mathbunnyru — DISMISSED** [review]
test_package is bigger than usual, but that's fine by me.

**Comment by ericriff** [comment]
> test_package is bigger than usual, but that's fine by me.

I know, I usually include a couple of headers and call a function. Just check for include dirs and linking. But I wanted to be consistent with what we have on 4.x.

This is a copy/pase of openCV official code.

**Comment by mathbunnyru** [comment]
> > test_package is bigger than usual, but that's fine by me.
> 
> I know, I usually include a couple of headers and call a function. Just check for include dirs and linking. But I wanted to be consistent with what we have on 4.x.
> 
> This is a copy/pase of openCV official code.

Oh, I see.
Yes, having consistent code is a good thing.
Thanks for the reply.

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
Great! Thank you!!

**Review by mathbunnyru — APPROVED** [review]

**Pull request merged**